server/src/main/java/org/opensearch/action/admin/cluster/remotestore/restore/RestoreRemoteStoreRequest.java

```diff
@@ -0,0 +1,211 @@
+        this.indices = indices;
+        return this;
+    }
+
+    /**
+     * Sets the list of indices that should be restored from the remote store
+     * <p>
+     * The list of indices supports multi-index syntax. For example: "+test*" ,"-test42" will index all indices with
+     * prefix "test" except index "test42". Aliases are not supported. An empty list or {"_all"} will restore all open
+     * indices in the cluster.
+     *
+     * @param indices list of indices
+     * @return this request
+     */
+    public RestoreRemoteStoreRequest indices(List<String> indices) {
+        this.indices = indices.toArray(new String[indices.size()]);
+        return this;
+    }
+
+    /**
+     * Returns list of indices that should be restored from the remote store
+     */
+    public String[] indices() {
+        return indices;
+    }
+
+    /**
+     * Specifies what type of requested indices to ignore and how to deal with wildcard expressions.
+     * For example indices that don't exist.
+     *
+     * @return the desired behaviour regarding indices to ignore and wildcard indices expression
+     */
+    public IndicesOptions indicesOptions() {
+        return indicesOptions;
+    }
+
+    /**
+     * Specifies what type of requested indices to ignore and how to deal with wildcard expressions.
+     * For example indices that don't exist.
+     *
+     * @param indicesOptions the desired behaviour regarding indices to ignore and wildcard indices expressions
+     * @return this request
+     */
+    public RestoreRemoteStoreRequest indicesOptions(IndicesOptions indicesOptions) {
+        this.indicesOptions = indicesOptions;
+        return this;
+    }
+
+    /**
+     * If this parameter is set to true the operation will wait for completion of restore process before returning.
+     *
+     * @param waitForCompletion if true the operation will wait for completion
+     * @return this request
+     */
+    public RestoreRemoteStoreRequest waitForCompletion(boolean waitForCompletion) {
+        this.waitForCompletion = waitForCompletion;
+        return this;
+    }
+
+    /**
+     * Returns wait for completion setting
+     *
+     * @return true if the operation will wait for completion
+     */
+    public boolean waitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * Parses restore definition
+     *
+     * @param source restore definition
+     * @return this request
+     */
+    @SuppressWarnings("unchecked")
+    public RestoreRemoteStoreRequest source(Map<String, Object> source) {
+        for (Map.Entry<String, Object> entry : source.entrySet()) {
+            String name = entry.getKey();
+            if (name.equals("indices")) {
+                if (entry.getValue() instanceof String) {
+                    indices(Strings.splitStringByCommaToArray((String) entry.getValue()));
+                } else if (entry.getValue() instanceof ArrayList) {
```

Comment:
Can it be ArrayList<non-String>?

Reply:
As indices is part of post body, it is better to get it as map from the `XContentParser`. This will help in easily adding new fields to the post body.